src/ellipse.rs

```diff
@@ -0,0 +1,261 @@
+        let Point { x: cx, y: cy } = new_center;
+        Ellipse {
+            inner: self.inner.set_translation(Vec2 { x: cx, y: cy }),
+        }
+    }
+
+    /// Returns the two radii of this ellipse.
+    ///
+    /// The first number is the horizontal radius and the second is the vertical radius, before
+    /// rotating by `x_rotation`.
+    pub fn radii(&self) -> Vec2 {
+        self.inner.svd().0
+    }
+
+    /// Set the radii of this ellipse.
+    #[must_use]
+    pub fn with_radii(self, new_radii: Vec2) -> Ellipse {
+        let rotation = self.inner.svd().1;
+        let translation = self.inner.get_translation();
+        Ellipse::private_new(translation, new_radii.x, new_radii.y, rotation)
+    }
+
+    /// The amount (in radians) that the ellipse should be rotated by (clockwise).
+    ///
+    /// This allows all possible ellipses to be drawn by always starting with an ellipse with the
+    /// two radii on the x and y axes.
+    pub fn x_rotation(&self) -> f64 {
+        self.inner.svd().1
+    }
+
+    /// Set the amount (in radians) that the ellipse should be rotated by (clockwise).
+    pub fn with_x_rotation(self, new_x_rotation: f64) -> Ellipse {
+        let scale = self.inner.svd().0;
+        let translation = self.inner.get_translation();
+        Ellipse::private_new(translation, scale.x, scale.y, new_x_rotation)
+    }
+}
+
+// We need to implement this ourselves as multiple affine maps represent the same ellipse. If `==`
+// on floats formed an equivalence relation (it doesn't), then the last parts of the SVD would form
+// an equivalence relation on affine maps.
+impl PartialEq for Ellipse {
+    fn eq(&self, other: &Self) -> bool {
+        self.inner.svd() == other.inner.svd()
+    }
+}
+
+impl Add<Vec2> for Ellipse {
+    type Output = Ellipse;
+
+    /// In this context adding a `Vec2` applies the corresponding translation to the eliipse.
+    #[inline]
+    fn add(self, v: Vec2) -> Ellipse {
+        Ellipse {
+            inner: Affine::translate(v) * self.inner,
+        }
+    }
+}
+
+impl Sub<Vec2> for Ellipse {
+    type Output = Ellipse;
+
+    /// In this context subtracting a `Vec2` applies the corresponding translation to the eliipse.
+    #[inline]
+    fn sub(self, v: Vec2) -> Ellipse {
+        Ellipse {
+            inner: Affine::translate(-v) * self.inner,
+        }
+    }
+}
+
+impl Mul<Ellipse> for Affine {
+    type Output = Ellipse;
+    fn mul(self, other: Ellipse) -> Self::Output {
+        Ellipse {
+            inner: self * other.inner,
+        }
+    }
+}
+
+impl From<Circle> for Ellipse {
+    fn from(circle: Circle) -> Self {
+        Ellipse::new(circle.center, Vec2::splat(circle.radius), 0.0)
+    }
+}
+
+impl Shape for Ellipse {
+    type BezPathIter = iter::Chain<iter::Once<PathEl>, ArcAppendIter>;
+
+    fn to_bez_path(&self, tolerance: f64) -> Self::BezPathIter {
+        // TODO is there a better way of doing this?
```

Comment:
Would it be better to convert a unit circle to a bezier and then apply the affine map?

Reply:
I'm happy with the code the way it is, mostly because it puts the error tolerance logic in a single place, namely `Arc::append_iter`. Sampling from the unit circle might allow some micro-optimizations, but I don't think it's worth it.

Another subtle point here: currently the arc logic is not as optimized as it might be in producing a minimal number of bezier segments as the ellipse goes more eccentric (it's based on the max of the major/minor axes). Applying affine to circle would lock us into not doing that, and if Arc ever is improved, again it would be best to have a single point.